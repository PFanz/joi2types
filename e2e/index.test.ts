import Joi from '@hapi/joi';
import joi2Types from '../dist';

test('normal', async () => {
  // example for react-router-config
  const schema = Joi.array().items(
    Joi.object({
      path: Joi.string().description('Any valid URL path'),
      component: Joi.string().description(
        'A React component to render only when the location matches.',
      ),
      redirect: Joi.string().description('navigate to a new location'),
      exact: Joi.boolean().description(
        'When true, the active class/style will only be applied if the location is matched exactly.',
      ),
    }).unknown(),
  );

  const types = await joi2Types(schema, {
    bannerComment: '/** comment for test */',
    interfaceName: 'IRoute',
  });
  expect(types.trim()).toMatchSnapshot();
});
