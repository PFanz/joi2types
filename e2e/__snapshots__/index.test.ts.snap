// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal 1`] = `
"/** comment for test */

export type IRoute = {
/**
 * Any valid URL path
 */
path?: string
/**
 * A React component to render only when the location matches.
 */
component?: string
/**
 * navigate to a new location
 */
redirect?: string
/**
 * When true, the active class/style will only be applied if the location is matched exactly.
 */
exact?: boolean
[k: string]: any
}[]"
`;
